src/socket.rs

```diff
@@ -27,12 +28,16 @@ pub struct UtpSocket<P> {
     cid_gen: Mutex<StdConnectionIdGenerator<P>>,
     accepts: mpsc::UnboundedSender<(Accept<P>, Option<ConnectionId<P>>)>,
     socket_events: mpsc::UnboundedSender<SocketEvent<P>>,
+    /// Maximum allowed connections, which only restricts outbound connections, but counts all
+    /// active connections.
+    max_conns: usize,
+    outbound_dropped: Arc<Notify>,
 }
 
 impl UtpSocket<SocketAddr> {
-    pub async fn bind(addr: SocketAddr) -> io::Result<Self> {
+    pub async fn bind(addr: SocketAddr, max_conns: usize) -> io::Result<Self> {
```

Comment:
Add a docstring with some hints about how to choose `max_conns`. Roughly:
- 100 seems like a reasonable starting place for a lot of use cases, 1k is probably a bit too high, depending on your setup, 10k will likely cause overload, 1-5 is probably so small that you won't be getting the full potential of the library
- A too-high value can cause connection timeouts, and various logs about packets taking too long to be processed locally
- A too-low value will cause connections to delay for a long time, while waiting for `connect()` or `connect_with_cid()` to return. You will not experience the full performance of the library